### PR TITLE
Aws download

### PR DESCRIPTION
**Pull request by acheronw**
AWS S3 download instead of http get requests for download_pages.

### CODE REVIEW

File: cc_corpus/download.py [hunk 1]
```diff
@@ -3,94 +3,121 @@
...
                                 offset-lengths pairs.
     :param retry_left: the number of retries left.
     :param delay_period: the base time unit for delay in seconds.
+    :param s3_download: whether to download via S3 or http.
+    :param session: S3 session to use. If None and we download via S3, a new
+                    session will be created.
     """
     logging.debug(
-        f'Downloading {len(offsets_and_lengths)=} ranges from {url}.')
+        f'Downloading {len(offsets_and_lengths)=} ranges from {url_or_key}.')
```
Comment (DavidNemeskey) [inline cc_corpus/download.py:43]: Didn't you say that you cannot download a list of ranges from s3? So either the function should raise `ValueError` `if s3_download and len(offsets_and_lengths) > 1` or it should call an s3-specific downloader function in a loop with a single range at a time.

File: scripts/download_pages.py [hunk 2]
```diff
@@ -17,28 +17,26 @@
...
 import threading
 import time
-from typing import TextIO
+from typing import Any, TextIO
 import zlib
 
 from cc_corpus.download import download_warc_ranges, DownloadError
 from cc_corpus.utils import notempty, num_digits, openall, otqdm
 
+BUCKET_NAME = 'commoncrawl'
```
Comment (DavidNemeskey) [inline scripts/download_pages.py:38]: We don't really need this, do we?

**Review by DavidNemeskey — CHANGES_REQUESTED** [review]
Does index downloading work? Is it done through http or s3?

**Comment by acheronw** [comment]
> Does index downloading work? Is it done through http or s3?

It works, and via http. Strangely enough common crawl haven't made them available via S3.

File: cc_corpus/download.py (hunk 1 above)
Comment (acheronw) [reply cc_corpus/download.py:43]: Done.

File: scripts/download_pages.py (hunk 2 above)
Comment (acheronw) [reply scripts/download_pages.py:38]: Right, in the end it was not used there.

**Review by DavidNemeskey — APPROVED** [review]

**Pull request merged**